packages/worker/src/argoWorker.ts

```diff
@@ -91,22 +90,32 @@ export function createWorkflow(run: PipelineRun): Workflow {
   };
 
   if (run.stages.some(s => s.volumes?.length || 0 > 0)) {
-    // TODO: replace global volume with volumes from benchmark
-    const volumeClaimTemplates: PersistentVolumeClaim[] = [
-      {
-        metadata: {
-          name: 'global',
-        },
-        spec: {
-          accessModes: ['ReadWriteOnce'],
-          resources: {
-            requests: {
-              storage: '1Gi',
-            },
-          },
-        },
-      },
-    ];
+    const volumeClaimTemplates: PersistentVolumeClaim[] = [];
+
+    for (const stage of run.stages) {
+      if (stage.volumes) {
+        for (const volume of stage.volumes) {
+          if (
+            !volumeClaimTemplates.some(c => c.metadata?.name === volume.name)
+          ) {
+            volumeClaimTemplates.push({
+              metadata: {
+                name: volume.name,
+              },
+              spec: {
+                // TODO: Set this as a parameter
```

Comment:
can we capture this in the SDS Tasks list backlog of this repo here?

Reply:
Inside of the project board you mean?

Reply:
actually i think this fits better in the tasks tab here in github rather than the project board for our other one